rapier 0.20 feature: `RevoluteJoint::angle` + more detailed joints enum (#530)
dimforge/bevy_rapier: CHANGELOG.md

and new features. Please have a look at the
[0.20 and 0.21 changelogs](https://github.com/dimforge/rapier/blob/master/CHANGELOG.md) of Rapier.**

### Modified

- Update to rapier `0.21`.
- Update to nalgebra `0.33`.

> - `ImpulseJoint` and `MultibodyJoint` now have a more detailed enum `TypedJoint` instead of a `GenericJoint`.

Reviewer: ```suggestion
- `ImpulseJoint::data` and `MultibodyJoint::data` are now a more detailed enum `TypedJoint` instead of a `GenericJoint`.
```

You can still access its inner `GenericJoint` with `.as_ref()` or `as_mut()`.

> - `data` fields from all joints (`FixedJoint`...) are now public, and their getters removed.

Reviewer: ```suggestion
- `data` fields from all joints (`FixedJoint`, …) are now public, and their getters removed.
```

### Added

> - `RapierContext::context.impulse_revolute_joint_angle` to compute the angle along a revolute joint’s principal axis.

Reviewer: ```suggestion
- Add `RapierContext::context.impulse_revolute_joint_angle` to compute the angle along a revolute joint’s principal axis.
```
